File: AutoShop/Volkswagen.m
//
//  Volkswagen.m
//  AutoShop
//
//  Created by Artem Makarov on 20.04.17.
//  Copyright © 2017 Artem Makarov. All rights reserved.
//

#import "Volkswagen.h"

@implementation Volkswagen 

- (id)init
{
    self = [super init];
    
    if (self) {
        self.volkswagenArray = [NSMutableArray array];
        self.descriptionCarDictionary = [NSMutableDictionary dictionary];
        self.modelArray = [[NSArray alloc] initWithObjects:@"Polo", @"Jetta", @"Passat", @"Tiguan", @"Touareg", nil];
    }
    
    return self;
}

- (void)addNewCar:(NewCarViewController *)controller volkswagenDictionary:(NSMutableDictionary *)dictionary {
    
    self.descriptionCarDictionary = dictionary;
    
    NSString *polo = @"https://d1hu588lul0tna.cloudfront.net/toyotaone/ruru/03-corolla-img_tcm-3020-716204.jpg";
    NSString *jetta = @"https://d1hu588lul0tna.cloudfront.net/toyotaone/ruru/1600-2_tcm-3020-949744.jpg";
    NSString *passat = @"https://d1hu588lul0tna.cloudfront.net/toyotaone/ruru/0001-rav4-full_tcm-3020-760795.jpg";
    NSString *tiguan = @"https://d1hu588lul0tna.cloudfront.net/toyotaone/ruru/027-lc-200-full-1_tcm-3020-669705.jpg";
    NSString *touareg = @"https://d1hu588lul0tna.cloudfront.net/toyotaone/ruru/02_tcm-3020-684645.jpg";
    
    
    if ([[self.descriptionCarDictionary objectForKey:@"CarModel"] isEqualToString:@"Polo"]) {
        [self.descriptionCarDictionary setObject:polo forKey:@"URLCar"];
        
    } else if ([[self.descriptionCarDictionary objectForKey:@"CarModel"] isEqualToString:@"Jetta"]) {
        [self.descriptionCarDictionary setObject:jetta forKey:@"URLCar"];
        
    } else if ([[self.descriptionCarDictionary objectForKey:@"CarModel"] isEqualToString:@"Passat"]) {
        [self.descriptionCarDictionary setObject:passat forKey:@"URLCar"];
        
    } else if ([[self.descriptionCarDictionary objectForKey:@"CarModel"] isEqualToString:@"Tiguan"]) {
        [self.descriptionCarDictionary setObject:tiguan forKey:@"URLCar"];
        
    } else if ([[self.descriptionCarDictionary objectForKey:@"CarModel"] isEqualToString:@"Touareg"]) {
        [self.descriptionCarDictionary setObject:touareg forKey:@"URLCar"];
        
    }
    
    [self.volkswagenArray addObject:dictionary];
    
}

@end

File: AutoShop(Swift)/Volkswagen.swift
//
//  Volkswagen.swift
//  AutoShop(Swift)
//
//  Created by Artem Makarov on 28.04.17.
//  Copyright © 2017 Artem Makarov. All rights reserved.
//

import Foundation

class Volkswagen: Auto, AddTableViewDelegate {

    let volkswagenArray = ["Polo", "Jetta", "Passat", "Tiguan", "Touareg"]
    
    let polo = "http://www.masmotors.ru/colors/volkswagen-polo/3.png"
    let jetta = "https://i.quto.ru/c533x400/4e3a4c8d925b4.png"
    let passat = "https://i.quto.ru/c533x400/4cef9b089c1df.jpeg"
    let tiguan = "https://i.quto.ru/c533x400/581b21057d3f3.jpeg"
    let touareg = "http://o.aolcdn.com/commerce/autodata/images/USC50VWS011B021001.jpg"
    
    func volkswagenDictionary(addTableViewController: AddTableViewController, carDictionary: [String:String]) {
        
        self.carDictionary = carDictionary
        
        if self.carDictionary["CarModel"] == "Polo" {
            self.carDictionary["URLCar"] = polo
            
        } else if self.carDictionary["CarModel"] == "Jetta" {
            self.carDictionary["URLCar"] = jetta
            
        } else if self.carDictionary["CarModel"] == "Passat" {
            self.carDictionary["URLCar"] = passat
            
        } else if self.carDictionary["CarModel"] == "Tiguan" {
            self.carDictionary["URLCar"] = tiguan
            
        } else if self.carDictionary["CarModel"] == "Touareg" {
            self.carDictionary["URLCar"] = touareg
            
        }
        
        self.carArray.append(self.carDictionary)
        print(self.carDictionary)
    }
    
}
